backend/FwLite/README.md

```diff
@@ -17,3 +17,58 @@ from repo root you will be able to develop the UI as well. From this folder it w
 ```bash
 task web-for-develop
 ```
+
+## High level project flow diagram
+
+```mermaid
+graph
+    subgraph Lexbox
+        Harmony[(Harmony)]
+        Mercurial[(Mercurial)]
+        sync("Sync FieldWorks lite \n (manually triggered)")
+        Harmony <---> sync
+        Mercurial <---> sync
+    end
+    subgraph User machine's/phone
+        FieldWorksLite[FieldWorks Lite]
+        ClassicFieldWorks[Classic FieldWorks]
+    end
+
+    FieldWorksLite <-- " Live Sync " --> Harmony
+    ClassicFieldWorks <-- " Send and Receive " --> Mercurial
+```
+
+### Adding a new Harmony change
+
+1. Create a change class.
+Reference [CreateComplexFormType](./LcmCrdt/Changes/CreateComplexFormType.cs) for a simple example of creating a new object.
+Reference [AddComplexFormTypeChange](./LcmCrdt/Changes/Entries/AddComplexFormTypeChange.cs) for an example of editing an existing object.
+2. Register that change class in [LcmCrdtKernel.ConfigureCrdt](./LcmCrdt/LcmCrdtKernel.cs)
+3. Add the change to [UseChangesTests.GetAllChanges](./LcmCrdt.Tests/Changes/UseChangesTests.cs)
+
+Things to watch out for
+
+#### any constructor parameter names must match the field/property they are assigned to
+Example: this will not work as the json serializer must match the property name with the argument name
```

Comment:
```suggestion
The JSON deserializer maps PascalCase property names to camelCase constructor parameter names. So, they need to be the same.

For example, this will not work:
```